src/Symfony/Component/DependencyInjection/ParameterBag/EnvPlaceholderParameterBag.php

```diff
@@ -62,9 +62,17 @@ public function getEnvPlaceholders()
 
     /**
      * Merges the env placeholders of another EnvPlaceholderParameterBag.
+     *
+     * @param EnvPlaceholderParameterBag $bag
      */
     public function mergeEnvPlaceholders(self $bag)
     {
-        $this->envPlaceholders = array_merge_recursive($this->envPlaceholders, $bag->getEnvPlaceholders());
+        $newPlaceholders = $bag->getEnvPlaceholders();
+
+        foreach ($newPlaceholders as $key => $newEntries) {
```

Comment:
I can see your point for most of those, but really using a 3 line  `foreach` to get the first element of an array seems like overkill. How about 
```
return reset($this->envPlaceholders[$env]); // return first result
```